src/main/java/com/boot/product/controller/ProductController.java

```diff
@@ -84,6 +84,16 @@ public ResponseEntity<ProductDTO> findProductBySlug(@Size(min = 3, max = 30, mes
         return new ResponseEntity<>(product, HttpStatus.OK);
     }
 
+    @GetMapping("/partialName/{partialName}")
+    @ResponseBody
+    public ResponseEntity<List<String>> findProductsByPartialName(@Size(min = 3, max = 30, message = "Product Name size has to be between 3 and 30 characters!") @PathVariable("partialName") String partialName)
```

Comment:
I wouldn't see the input string as a product name so maybe the error message might not be accurate

Reply:
well, its a partial product name..
Any suggestions? I couldn't find a better one :)

Reply:
Query string? or Partial product name?